2015-11-01    Jiachen Liu    jasonliujc@ucla.edu
           *cast the diff variable to signed signed long long int
           in order to prevent integer overflow
====================================================================================
--- timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ newtime.h	2015-02-08 20:58:04.100847000 -0800
@@ -45,8 +45,14 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  
+  signed long long test=(signed long long int)a.tv_sec - (signed long long int)b.tv_sec;
+  /* I did two longs is because the we want it to be 64 bits. */
+  if (test>=0)
+     return test>0;
+  else 
+     return -1; 
+  
 }
 
 # if ! HAVE_DECL_NANOSLEEP
